### ISSUE TITLE
Grants Identifying Changes/Modify When No Changes Made

### ISSUE THREAD

**Issue by willwooten** [Snowflake-Labs/terraform-provider-snowflake#1877]
**Provider Version**

0.66.1

**Terraform Version**

1.4.6

**Describe the bug**

1) snowflake_database_grant resource which grants "IMPORTED PRIVILEGES" is making modifications when no change was made to resource. enable_multiple_grants is set to true.
2) snowflake_tag resource is identifying changes when no changes were made.

**Expected behavior**

If no change is made to the resource, i'd expect no changes in the terraform plan/apply

**Code samples and commands**

Code in tf files:
```
resource "snowflake_database_grant" "grant_imported_privileges_to_snowflake_database" {
  database_name          = "SNOWFLAKE"
  privilege              = "IMPORTED PRIVILEGES"
  roles                  = [snowflake_role.examplerole.name, snowflake_role.examplerole2.name]
  with_grant_option      = false
  enable_multiple_grants = true
}
```

```
resource "snowflake_tag" "cost_usage_dev_schema_business_line_tag" {
  name           = "BUSINESS_LINE"
  database       = snowflake_database.dev.name
  schema         = snowflake_schema.cost_usage_dev.name
  allowed_values = ["Value1", "Value2"]
}
```

What we get when run terraform plan:

```
  # snowflake_tag.cost_usage_dev_schema_business_line_tag will be updated in-place
  ~ resource "snowflake_tag" "cost_usage_dev_schema_business_line_tag" {
      ~ allowed_values = [
          - "Value2",
            "Value1",
          + "Value2",
        ]
        name           = "BUSINESS_LINE"
        # (2 unchanged attributes hidden)
    }
```

```
  ~ resource "snowflake_database_grant" "grant_imported_privileges_to_datadog_role" {
        id                     = "SNOWFLAKE|IMPORTED PRIVILEGES|false|DATADOG_ROLE,DATADOG_ORGADMIN_ROLE|"
      ~ roles                  = [
          + "examplerole1",
          + "examplerole2",
        ]
        # (5 unchanged attributes hidden)
    }
```

**Additional context**

This problem only occured after update to 0.66

**Comment by awittzb** [comment]
I am seeing the same behavior after updating to 0.66.1 (from 0.37.0, so, I cannot narrow down when exactly it was introduced).

When I apply the changes, terraform reports that the changes were made but a subsequent `terraform plan` shows the _same_ changes still ready to be made.

**Provider Version**

0.66.1

**Terraform Version**

1.3.7

**Comment by toni-moreno** [comment]
As I explained in https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/1868#issuecomment-1592754879 ,  I have the same problem with 0.66.X provider version with role grants.  I can not do new changes  because it will destroy valid snowflake users on my roles.



It is adding a new parameter `enable_multiple_grants = true`  and also trying to remove some of the users belonging to the role. This is a very big trouble for us

```
  # snowflake_role_grants.role_grant["MY_ROL"] will be updated in-place
  ~ resource "snowflake_role_grants" "role_grant" {
      + enable_multiple_grants = true
        id                     = "MY_ROL||XXXXXXX,YYYYYYYY,ZZZZZZZ"
      ~ users                  = [
          - "XXXXXXX",
          - "YYYYYYY",
          - "ZZZZZZZ"
        ]
        # (2 unchanged attributes hidden)
    }


Any idea on how to continue working with terraform and snowflake??

**Comment by rjoelnorgren** [comment]
Experiencing the same issue with the provider attempting to always recreate the imported privileges grant on the SNOWFLAKE database with 0.67.

**Comment by philippeboyd** [comment]
bug still persists with `v0.68.1`

**Comment by denzhel** [comment]
Still happens in v0.85.0

**Comment by sfc-gh-jcieslak** [comment]
I think the final fix was provided in 0.87.1, please see and let me know if we can close this issue as it's pretty much the same as https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2366.

**Comment by sfc-gh-jcieslak** [comment]
Closing due to long inactivity and because it mainly refers to the deprecated grant resource no longer available in the latest provider versions. Please migrate to the newer counterparts and in case any issues with them create a new gh issue.
Thank You.
